### PR TITLE
header32

### PR DESCRIPTION
**Pull request by emirhandumann**

### CODE REVIEW

File: editor-components/header/header32/header32.tsx [hunk 1]
```diff
@@ -1,76 +1,219 @@
...
-  title: string;
-  image: string;
-  description: string;
-  button: IButton;
-};
+type Card = {
+  backgroundImage: string;
+  imageTitle: string;
+  imageDescription: string;
+  urlTitle: string;
```
Comment (dogusmiuw) [inline editor-components/header/header32/header32.tsx:16]:
```suggestion
  imageTitle: JSX.Element;
  imageDescription: JSX.Element;
  urlTitle: JSX.Element;
```

File: editor-components/header/header32/header32.tsx [hunk 2]
```diff
@@ -1,76 +1,219 @@
...
 
-type IButton = {
-  buttonText: string;
-  buttonClick: string;
+type Button = {
+  leftIcon: string;
+  rightIcon: string;
 };
 
+type AppendDotsFunction = NonNullable<Settings["appendDots"]>;
```
Comment (dogusmiuw) [inline editor-components/header/header32/header32.tsx:25]: remove

File: editor-components/header/header32/header32.tsx [hunk 3]
```diff
@@ -1,76 +1,219 @@
...
+        {
+          type: "object",
+          key: "sliderContent",
+          displayer: "Slider Content",
+          value: [
+            {
+              type: "image",
+              key: "backgroundImage",
+              displayer: "Background Image",
+              value: "https://themes.pixelwars.org/archy/demo-01/wp-content/uploads/sites/2/2023/05/daniel-stone-A4TtfFo_qRw-unsplash.jpg",
```
Comment (dogusmiuw) [inline editor-components/header/header32/header32.tsx:82]: should be spica storage url

File: editor-components/header/header32/header32.tsx [hunk 4]
```diff
@@ -80,53 +223,117 @@ class Header32 extends BaseHeader {
...
+                      <div
+                        className={this.decorateCSS("content")}
+                        style={{ backgroundImage: `url(${item.backgroundImage})` }}
+                      >
+                        <div className={this.decorateCSS("content-inner")}>
+                          <div className={this.decorateCSS("slide-shape")}>
+                          </div>
+
+                          <h2 className={this.decorateCSS("image-title")}>
+                            {item.imageTitle}
```
Comment (dogusmiuw) [inline editor-components/header/header32/header32.tsx:270]: it its empty, do not render relevant elements

File: editor-components/header/header32/header32.tsx [hunk 5]
```diff
@@ -80,53 +223,117 @@ class Header32 extends BaseHeader {
...
+                          </h2>
+
+                          <div className={this.decorateCSS("image-details")}>
+                            <p className={this.decorateCSS("description")}>
+                              {item.imageDescription}
```
Comment (dogusmiuw) [inline editor-components/header/header32/header32.tsx:275]: it its empty, do not render relevant elements

File: editor-components/header/header32/header32.tsx [hunk 6]
```diff
@@ -80,53 +223,117 @@ class Header32 extends BaseHeader {
...
+
+                            <div className={this.decorateCSS("stick")}></div>
+
+                            <div className={this.decorateCSS("url-container")}>
+                              <ComposerLink
+                                key={`hdr-32-${index}`}
+                                path={item.url}
+                              >
+                                <p className={this.decorateCSS("url-title")}>
+                                  {item.urlTitle}
```
Comment (dogusmiuw) [inline editor-components/header/header32/header32.tsx:286]: it its empty, do not render relevant elements

File: editor-components/header/header32/header32.tsx [hunk 7]
```diff
@@ -80,53 +223,117 @@ class Header32 extends BaseHeader {
...
+
+                          <div className={this.decorateCSS("nav-buttons")}>
+                            <button
+                              className={this.decorateCSS("nav-button")}
+                              onClick={() => {
+                                this.getComponentState("slider-ref").current.slickPrev();
+                              }}
+                            >
+                              <ComposerIcon
+                                name={this.getPropValue("leftButton")[0].value}
```
Comment (dogusmiuw) [inline editor-components/header/header32/header32.tsx:301]: do not access values like this: ...[0].value...

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
please handle the changes

**Comment by alperensen** [comment]
Updated header32

File: editor-components/header/header32/header32.module.scss [hunk 8]
```diff
@@ -1,335 +1,383 @@
...
+  from {
+    transform: translateX(80%);
+    opacity: 0;
+  }
+  to {
+    transform: translateY(0);
+    opacity: 0.7;
+  }
+}
+
```
Comment (Ayazelar) [inline editor-components/header/header32/header32.module.scss:35]: there is the same animation named descriptionBottom. delete this one

File: editor-components/header/header32/header32.module.scss [hunk 9]
```diff
@@ -1,335 +1,383 @@
...
-                .background-image {
-                  img {
-                    transform: scale(1.0);
+                p{
+                  >blinkpage {
+                    display: -webkit-box !important;
+                    -webkit-line-clamp: 3;
+                    -webkit-box-orient: vertical;
+                    overflow: hidden;
+                    text-overflow: ellipsis;
```
Comment (Ayazelar) [inline editor-components/header/header32/header32.module.scss:155]: check mixin and include in scss

File: editor-components/header/header32/header32.tsx [hunk 10]
```diff
@@ -22,111 +25,391 @@ class Header32 extends BaseHeader {
...
+          this.setComponentState("activeSlide", next);
+        }
+      },
+      dotsClass: this.decorateCSS("dots"),
+      customPaging: function (i: number) {
+        return (
+          <div>
+            <button></button>
+            <span className={styles.dotIndex}>{`0${i + 1}`}</span>
+          </div>
```
Comment (Ayazelar) [inline editor-components/header/header32/header32.tsx:258]: it looks like the button and span seem unnecessary. also, use decorateCss in classNames

**Review by Ayazelar — CHANGES_REQUESTED** [review]
change

**Pull request merged**